### DISCUSSION TITLE
CustomTube discontinuation

### DISCUSSION THREAD

**Discussion by lightbeam24** [lightbeam24/CustomTube#240] [Announcements]
Tldr: CustomTube has been discontinued. My current project, StarTube, is the successor, and it requires Vorapis 2013 (V3) to function.
Download V3: https://vorapis.pages.dev/#/
Download StarTube: https://greasyfork.org/en/scripts/485622-startube
V3 Github Link: https://github.com/VORAPIS/v3 (Note that V3 is not open source, this page is mainly for issue tracking)

There is also a beta version of StarTube for those that want it: https://greasyfork.org/en/scripts/500060-startube-beta

-----

Guess it's about time I make an official statement regarding this. At first I genuinely wasn't sure if I wanted to discontinue this project, but it has since become clear that it's the only correct option. I did intend to address it eventually, but I was originally going to wait until StarTube (my current project) was in a more ideal state. Perhaps that wasn't the right move.

Unfortunately, I am no longer working on CustomTube. There is a good reason for this however. As you might be aware, a much better project has since released, Vorapis 2013 (or V3). V3 is a custom frontend/client that essentially revives 2013/2014 YouTube in its entirety, using the original HTML instead of styling YouTube's current HTML. It's much faster than the regular YouTube website, and most importantly, **is much easier to work with**. Since it uses 2013 HTML, not only is the HTML structure easier to work with than regular YouTube, but I can trust that it won't randomly change one day (which happens A LOT with normal YouTube).

Which is why I started working on StarTube, which originally was just an enhancement script for V3, but evolved into basically being CustomTube for V3. It currently allows you to use the 2011-2022 layouts, which isn't as many as CustomTube, but I've been able to add those layouts way faster than I did with CT. The layouts are also far more accurate than they were in CT (2019-2022 might be an exception, for now anyway). Also, there are some extra layouts in there that weren't in CT, such as Epic Panda (a prototype layout that was tested in mid-late 2012), and 2016 Polymer (though that one is missing key features right now).

I was originally gonna continue CustomTube despite V3 being out, but I quickly realized it was a waste of effort. Regular YouTube is a major pain to work with, and no matter what I added to CustomTube, it would still fall far short of what's possible on V3. And then YouTube would just break it in a few weeks anyway.

I've brought some features from CustomTube over to StarTube, namely medium related video size, and more importantly playlist timestamps:

![Screenshot_2024-08-01_11-51-30](https://github.com/user-attachments/assets/b41d2031-3508-4782-9c7b-9b48faf9cb76)

But I've also been able to implement features that I never got to implement in CustomTube, like Alt-Watch9, my custom watch page layout concept:

![Screenshot_2024-08-01_11-53-55](https://github.com/user-attachments/assets/9a459610-0bf1-4047-89d8-631a51946535)

The only real problem is that due to the way StarTube is designed, while it's far more customizable than CT, it's also a lot more complicated to configure. This has been mitigated with layout presets, but any further customization is bound to be very daunting for many people. I'm trying to think of ways to fix this, but I haven't quite figured it out yet. StarTube is also overall less stable than CustomTube (when it was still functioning well anyway). It will also likely go through major changes yet.

CustomTube will only continue to break down more and more. I just tried it today for the first time in ages, and was met with a completely broken homepage. It's not feasible to maintain unless you're super invested in it. StarTube is already better than CT in most ways, and will only continue to get better. For that and all the reasons stated above, I encourage everyone to switch to V3, with or without StarTube installed.

Download V3: https://vorapis.pages.dev/#/
Download StarTube: https://greasyfork.org/en/scripts/485622-startube
V3 Github Link: https://github.com/VORAPIS/v3 (Note that V3 is not open source, this page is mainly for issue tracking)

There is also a beta version of StarTube for those that want it: https://greasyfork.org/en/scripts/500060-startube-beta

Here are a few screenshots of what V3 + StarTube can do:

![Screenshot_2024-08-01_11-33-16](https://github.com/user-attachments/assets/5552f37a-322b-4b0e-91b7-6f3a55f13c53)
![Screenshot_2024-08-01_11-36-48](https://github.com/user-attachments/assets/9c198b95-e725-4c2a-a16f-d4d52a9573d1)
![Screenshot_2024-08-01_12-05-32](https://github.com/user-attachments/assets/4ce99d26-d607-4d94-885f-101585d2cea7)
![Screenshot_2024-08-01_11-37-26](https://github.com/user-attachments/assets/15cd05e9-9fc3-419e-b4fc-d9aa0395aa51)
![Screenshot_2024-08-01_11-37-41](https://github.com/user-attachments/assets/9fd5cb66-a041-40d9-89b3-5ab9af96a971)
![Screenshot_2024-08-01_11-38-45](https://github.com/user-attachments/assets/6098ee39-69f8-48ff-82be-3ac23d19aef0)
![Screenshot_2024-08-01_11-39-13](https://github.com/user-attachments/assets/2d0a010f-b82a-4d7b-b6dc-33bf36a3a835)
![Screenshot_2024-08-01_11-39-46](https://github.com/user-attachments/assets/6f559c36-2374-4226-9c5f-a7e046df633b)
![Screenshot_2024-08-01_11-40-10](https://github.com/user-attachments/assets/b28ca634-6e96-45e2-bfea-3f98ffe6835a)
![Screenshot_2024-08-01_11-40-31](https://github.com/user-attachments/assets/6d671098-b5a5-4218-8ff1-f746cc175ca9)
![Screenshot_2024-08-01_11-41-34](https://github.com/user-attachments/assets/d8a4c874-3b52-4248-b081-b6080e5e7e2e)

![Screenshot_2024-08-01_11-41-39](https://github.com/user-attachments/assets/0311b4a1-da23-4060-8ea6-3abd5d907440)
![Screenshot_2024-08-01_11-41-43](https://github.com/user-attachments/assets/c45ec1f9-9e06-46d8-ba78-5b17966ff67b)
![Screenshot_2024-08-01_11-42-22](https://github.com/user-attachments/assets/bb740491-d690-40cd-ab07-839cad87f559)
![Screenshot_2024-08-01_11-43-23](https://github.com/user-attachments/assets/fb4f4dfe-6afa-4af4-87fa-e99ced04158b)
![Screenshot_2024-08-01_11-44-46](https://github.com/user-attachments/assets/3bc57f65-f12c-4fe5-9684-ee5d16811900)

**Comment by userunknown358** [comment]
You can add the "2017 layout enabled" thing on the images so we know which one is which. Also, you can put a message on CT saying it is now discontinued on the YouTube home page when someone installs it.

But yeah, I've known CT ever since January 2024, it wasn't that accurate.

**Reply by lightbeam24** [reply]
> You can add the "2017 layout enabled" thing on the images so we know which one is which.

Too much work for just this post, and it didn't even cross my mind at the time.

> Also, you can put a message on CT saying it is now discontinued on the YouTube home page when someone installs it.

I am planning to do something like that yet, but I need to think carefully about it so I do it in the best way possible. Ideally the banner would include links to V3 and StarTube, but I'd have to look into if that's even allowed by the extension stores. So for now, I won't do any of that.

**Comment by gameshowrecaps** [comment]
Thank you for working on this extension when you could. It has been a blast using this extension throughout the last year up to now. I look forward to the progress of StarTube.

**Reply by userunknown358** [reply]
It was fun using CustomTube a month before it's final release and the final beta (Beta 1.3.0) But now the owner has ceased operations of CustomTube in favour of V3 + StarTube. I'm also looking forward to seeing it more and more accurate up to the point it's like living in a certain year again. 


In loving memory of CustomTube
2023-2024

**Comment by coders22web** [comment]
Hello! I do admire the new successor to CustomTube, but I just have one question. When I was using the "Aozora 2011," "Cosmic Panda," "Epic Panda," "Early 2013," and "Early 2013 Alt" presets, I encountered numerous problems. I couldn't see any shorts or livestream VODs on the YouTube channel pages. They don't even show up as normal videos. It's also missing the external links in the "About" section, like "Facebook," "Twitter," etc., as well as the "Other Channels" section, mainly for channels that have only one other channel featured in that section.

Here are some examples:

With my channel: https://www.youtube.com/channel/UCgFGrc9jo45tW0wuJwn5XyA

2013 About Page:
![Screenshot 2024-08-22 015315](https://github.com/user-attachments/assets/ae4ced18-660c-4b7f-b1ce-a922dd6d7361)

2012 About Page (On the right side, notice the links [The "My Carrd" link] are missing, as well as the "Other Channels" section [Which I named "Secondary Channel"], mainly since I have only one channel featured in that section):
![image](https://github.com/user-attachments/assets/dd542889-d19e-4ce7-8d5a-376e76c9bda8)

This is an example of what the "About" section should look like, it has all these links to "Facebook," "Twitter," etc.
![image](https://github.com/user-attachments/assets/1c6f1b84-40ce-4051-bf2a-34e50f99eecd)
Taken from this video: https://www.youtube.com/watch?v=a0uUM2FHLdw

2013 Videos Page:
![Screenshot 2024-08-22 015405](https://github.com/user-attachments/assets/867e7e0d-d3f6-493c-b69a-f5f782b9aacc)

2012 Videos Page (Notice how the shorts are missing and never shown, not even as normal videos, also the case for livestream VODs):
![Screenshot 2024-08-22 015451](https://github.com/user-attachments/assets/18bada55-ff33-4979-84aa-64d14a19330d)

I've also encountered another problem regarding "playlist" sections:

2013 Playlist Section:
![image](https://github.com/user-attachments/assets/1cfb6687-55e9-411d-a7b0-b1159884a824)

2012 Playlist Section (Notice how none of my playlists are showing):
![image](https://github.com/user-attachments/assets/e436b643-5fcb-4838-b712-ce03d1a32575)

For "Topic" channels, mainly ones from music artists, their "Videos" sections are missing "Art Tracks" (https://support.google.com/youtube/answer/6007071). 

With this channel: https://www.youtube.com/channel/UC0NqJ6MhRnkRaWAmBgyyjTA/featured

2013 Topic Channel Home Page:
![image](https://github.com/user-attachments/assets/30d60ae1-eabe-4c44-bd6b-cd79751fb831)

2012 Topic Channel Home Page (The "Albums & Singles" section is there on the right, the album playlists from there are the only places to look for "Art Tracks" via channel pages):
![image](https://github.com/user-attachments/assets/a74c438c-bb50-44ba-8411-aa24dbc9745e)

2013 Topic Channel Videos Page:
![image](https://github.com/user-attachments/assets/91b286f4-40f9-4d6f-8e34-6f03ffde1c01)

2012 Topic Channel Videos Page (The "Art Tracks" are nowhere to be found):
![image](https://github.com/user-attachments/assets/279089e5-11ec-44a8-96c1-581f487703ab)

Is there any way to fix this? Is there a setting for that?

**Reply by home-gihub** [reply]
Not open source. not using

**Comment by SatinFoil** [comment]
Can you please work on the accuracy for the 2019-2020 polymer layouts (mostly 2020), as customtube beta's polymer layouts was far more accurate than startube's and I feel I've been downgraded for those ones.

**Reply by userunknown358** [reply]
The early Polymer layouts are being worked on at the moment I'm commenting so expect accuracy of the early Polymer layouts in the next version of StarTube.

>customtube beta's polymer layouts was far more accurate than startube

Not really, accounts switching, channel pages are not accurate and it lacks the apps icon that was there back in late 2019

**Reply by SatinFoil** [reply]
> The early Polymer layouts are being worked on at the moment I'm commenting so expect accuracy of the early Polymer layouts in the next version of StarTube.
> 
> > customtube beta's polymer layouts was far more accurate than startube
> 
> Not really, accounts switching, channel pages are not accurate and it lacks the apps icon that was there back in late 2019

Ah, alright
Also, for the accuracy, I was mostly talking about the watch page. Also, I don't see the apps button even on startube? Or did you mean something else?

**Comment by Andrew2592009** [comment]
Hey @lightbeam24, i wonder if StarTube (the successor to CustomTube) will have the 2008 and 2009 layout?

**Reply by lightbeam24** [reply]
Sorry for the late response.

The answer is: I want to implement them eventually, but improving the other layouts is more of a priority. As always, nothing is guaranteed.

**Comment by NewSummo** [comment]
Constant issue with Startube:
0 Error count
expand
Ni: KOR VERSION: 0.8.8 (Rev. 14541.1599 vgb_service_perf_obtfarewell_out4)
Refuses to play ANY video

**Reply by userunknown358** [reply]
Please report this to V3's official Discord server, we don't do reports here.

**Comment by jmays-12** [comment]
i installed startube +v3 and it just is so bloated and confusing compared to customtube. customtube has worked flawlessly and is so easy to use compared to this new mess. 800000 advanced options and i cant figure out how to get all these ugly ui elements working like the 2019 clean set up i had with customtube. god i hope someone forks off customtube because this other crap sucks

**Reply by lightbeam24** [reply]
First of all, make sure you're using the *2018* layout, as that is StarTube's equivalent to CustomTube's 2019 layout (really, I've since realized it should've been called 2018 in CustomTube all along).

Second, ignore the advanced options. The advanced options are only still there because originally StarTube ONLY had those advanced options, so I couldn't get rid of them without reworking the whole project as well as sacrificing potential customization for those who are really into it. The advanced options have been pushed to the side in recent updates, especially 2.2.0. I was under the impression that as of 2.2.0, StarTube was finally easy to use (it used to be much worse). But perhaps I'm wrong. Any suggestions to make it easier?

> i cant figure out how to get all these ugly ui elements working like the 2019 clean set up i had with customtube.

There's probably nothing to figure out. StarTube is designed so that you don't really have to change any settings other than the layout selector (and maybe the player size). I think the issues you're having are due to StarTube's 2018 & 2019 layouts being inaccurate. I think CustomTube at its peak probably had more accurate 2018 & 2019 layouts than StarTube has, at least for now.

**Reply by VelvetShadows** [reply]
Is there some conflict with my setup or do playlists of others that you have saved not show up on StarTube?

ie. I can save https://www.youtube.com/playlist?list=PLj_Goi54wf0fry2K9Rry1UqzGjnId01Bv that playlist and it will show up among my playlists as you'd expect with YouTube standard, Custom Tube or YouTube Redux but in StarTube it's hidden.

**Reply by lightbeam24** [reply]
> Is there some conflict with my setup or do playlists of others that you have saved not show up on StarTube?
> 
> ie. I can save https://www.youtube.com/playlist?list=PLj_Goi54wf0fry2K9Rry1UqzGjnId01Bv that playlist and it will show up among my playlists as you'd expect with YouTube standard, Custom Tube or YouTube Redux but in StarTube it's hidden.

Hmm... looks like you are correct. https://www.youtube.com/feed/playlists doesn't seem to work in V3. There's https://www.youtube.com/feed/library, but the only way to access it is to use one of the 2013 layouts and enable GUIDE_USER_ITEM_GO_TO_LIBRARY in V3's settings. Also, it's just a shelf of playlists instead of a grid, so it's not as convenient.

**Comment by IMNdi** [comment]
I just wanted to have the subscriptions feed not be 3 tiles, but 5 or 6. 

Now the addon is deprecated silently, it still opens the changelog every browser update (but can't notify it's deprecated), Google says it's now a script and runs on an addon that has to be sideloaded, is closed, and has me install a Discord client and join a server. 

And best of all, neither of these two have a configurable videos per row. Excellent. Nor a dark mode (that works). Also, they break all other Youtube addons. 

But hey, if you disagree, join my Telegram.

I know this is mostly a rant. Nobody is forcing me to user StarFire, I can just take home the fact that CustomTube is abandoned. Fair enough. As kids say nowadays, F.

**Reply by bangbangyouall** [reply]
"I just wanted to have the subscriptions feed not be 3 tiles, but 5 or 6." That's.. oddly specific.

"Now the addon is deprecated silently, it still opens the changelog every browser update (but can't notify it's deprecated), Google says it's now a script and runs on an addon that has to be sideloaded, is closed, and has me install a Discord client and join a server." Eh??? You don't have to use bumfucking Discord for it, you can simply install it on GreasyFork with Violentmonkey. You can also simply just go to the Vorapis site, if that's what you're talking about.

"And best of all, neither of these two have a configurable videos per row. Excellent." How is this a criticism? That's so damn specific, you should not care about that!

"Nor a dark mode (that works)." Deal with it.

"Also, they break all other Youtube addons." There's userscript equivalents to those addons. I even have a functioning script equivalent for SponsorBlock installed.

"But hey, if you disagree, join my Telegram." How is that relevant? Are you trying to groom them?

"I know this is mostly a rant. Nobody is forcing me to user StarFire," Nice job butchering that spelling.

"I can just take home the fact that CustomTube is abandoned. Fair enough. As kids say nowadays, F." More like a W in.. literally everybody else's book. StarTube has more layouts, more accuracy, it's lots of fun! Wouldn't hurt to try!


In conclusion, StarTube is bad because I can't handle looking at small rows of videos & I cannot do my research.

**Reply by userunknown358** [reply]
StarTube is still in development right now, it's best to be patient. If you want high accuracy, you can stick with rehike

**Reply by bangbangyouall** [reply]
StarTube has dark mode now, BABY!

**Comment by home-gihub** [comment]
It's not open source though

**Reply by FlashlightET** [reply]
![image](https://github.com/user-attachments/assets/37f6fe9e-918d-4a54-9b8d-7ffdbe77c37f)

**Reply by home-gihub** [reply]
yes but it needs Vorapis 2013 --__--

**Comment by gameshowrecaps** [comment]
Still using CT after the discontinuation, and not surprised at the things breaking slowly. Today, the videos on the right side/related videos broke big time. I don't think I'll use the new StarTube actually. If things break majorly which it may by the end of the year, I may have to uninstall the extension for good.

Edit: I don't see this issue on my alt, YouTube may have started A/B testing something. It's unknown what. 

![Screenshot 2024-11-02 182740](https://github.com/user-attachments/assets/9c7184ae-5182-4ae8-a598-2ebeb4494a8c)

**Reply by bangbangyouall** [reply]
Why are you still reluctant to use StarTube even after it literally got better in every way?

**Reply by gameshowrecaps** [reply]
Because I have every right to use it over StarTube. And also, it doesn't really work well with my pitch changer extension I use ([screwmycode.in](https://chromewebstore.google.com/detail/screwmycodein-pitch-contr/lnoedjelpkmhegefhhjljmggjdicnbaa?hl=en)). In addition, I always had errors after a couple seconds of videos. Last I checked StarTube, videos errored.

**Reply by bangbangyouall** [reply]
Ah, I see. Maybe see if Lightbeam can fix it, though it might be a lost cause for your extension.

**Reply by gameshowrecaps** [reply]
Yeah unfortunately that may very well be the case. I think the error thing on StarTube is fixed last I checked recently. I may wait a bit longer to fully use StarTube.

**Comment by ikkqqstudio** [comment]
But it lacks System theme which I needed for Auto dark mode
